website/static/js/registrationUtils.js

```diff
@@ -0,0 +1,764 @@
+    self.draft(draft);
+};
+RegistrationEditor.prototype.submitForReview = function() {
+    var self = this;
+
+    var draft = self.draft();
+    var metaSchema = draft.metaSchema;
+    var messages = metaSchema.messages;
+    var beforeSubmitForApprovalMessage = messages.beforeSubmitForApproval || '';
+    var afterSubmitForApprovalMessage = messages.afterSubmitForApproval || '';
+
+    bootbox.confirm(beforeSubmitForApprovalMessage, function(confirmed) {
+        if (confirmed) {
+            $osf.postJSON(self.urls.submit.replace('{draft_pk}', self.draft().pk), {}).then(function() {
+                bootbox.dialog({
+                    closeButton: false,
+                    message: afterSubmitForApprovalMessage,
+                    title: 'Pre-Registration Prize Submission',
+                    buttons: {
+                        registrations: {
+                            label: 'Return to registrations page',
+                            className: 'btn-primary pull-right',
+                            callback: function() {
+                                window.location.href = self.draft().urls.registrations;
+                            }
+                        }
+                    }
+                });
+            }).fail($osf.growl.bind(null, 'Error submitting for review', language.submitForReviewFail));
+        }
+    });
+};
+RegistrationEditor.prototype.submit = function() {
+    var self = this;
+    var currentNode = window.contextVars.node;
+    var currentUser = $osf.currentUser();
+    var messages = self.draft().messages;
+    bootbox.confirm(messages.beforeSubmitForApproval, function(result) {
+        if (result) {
+            $osf.postJSON(self.urls.submit.replace('{draft_pk}', self.draft().pk), {
+                node: currentNode,
+                auth: currentUser
+            }).then(function() {
+                bootbox.dialog({
+                    message: messages.afterSubmitForApproval,
+                    title: 'Pre-Registration Prize Submission',
+                    buttons: {
+                        registrations: {
+                            label: 'Return to registrations page',
+                            className: 'btn-primary pull-right',
+                            callback: function() {
+                                window.location.href = self.draft().urls.registrations;
+                            }
+                        }
+                    }
+                });
+            }).fail($osf.growl.bind(null, 'Error submitting for review', language.submitForReviewFail));
+        }
+    });
+};
+/**
+ * Create a new draft
+ **/
+RegistrationEditor.prototype.create = function(schemaData) {
+    var self = this;
+
+    var metaSchema = self.draft().metaSchema;
+
+    return $osf.postJSON(self.urls.create, {
+        schema_name: metaSchema.name,
+        schema_version: metaSchema.version,
+        schema_data: schemaData
+    }).then(self.updateData.bind(self));
```

Comment:
Should there also be a `fail` handler here?

Also, all methods that do http work should return a promise.